library/classes/Prescription.class.php

```diff
@@ -426,23 +426,31 @@ function set_medication($med)
 
         //check if this drug is on the medication list
         $dataRow = sqlQuery("select id from lists where type = 'medication' and activity = 1 and (enddate is null or cast(now() as date) < enddate) and upper(trim(title)) = upper(trim('" . add_escape_custom($this->drug) . "')) and pid = '" . add_escape_custom($this->patient->id) . "' limit 1");
+        //Get dosage instructions
+        $dosageInstructions = $this->dosage . " in " . $this->form_array[$this->form] . " form " . $this->interval_array[$this->interval] . "(" . $this->size . " " . $this->unit_array[$this->unit] . ")";
 
         if ($med && !isset($dataRow['id'])) {
             $dataRow = sqlQuery("select id from lists where type = 'medication' and activity = 0 and (enddate is null or cast(now() as date) < enddate) and upper(trim(title)) = upper(trim('" . add_escape_custom($this->drug) . "')) and pid = '" . add_escape_custom($this->patient->id) . "' limit 1");
 
             if (!isset($dataRow['id'])) {
                 //add the record to the medication list
                 sqlStatement("insert into lists(date,begdate,type,activity,pid,user,groupname,title) values (now(),cast(now() as date),'medication',1,'" . add_escape_custom($this->patient->id) . "','" . add_escape_custom($_SESSION['authUser']) . "','" . add_escape_custom($_SESSION['authProvider']) . "','" . add_escape_custom($this->drug) . "')");
+                // Also ensure it gets added to lists_medication
+                $medListId = sqlQuery("select id from lists where type = 'medication' and (enddate is null or cast(now() as date) < enddate) and upper(trim(title)) = upper(trim('" . add_escape_custom($this->drug) . "')) and pid = '" . add_escape_custom($this->patient->id) . "' limit 1");
+                sqlStatement("insert into lists_medication(list_id, drug_dosage_instructions) values ('" . add_escape_custom($medListId["id"]) . "', '" . add_escape_custom($dosageInstructions) . "')");
```

Comment:
No, I double-checked that if a prescription is not selected to be on the med list, it will not appear. If you look at line 437, which is the current, working functionality, that's only ran when a provider wishes to add it.

Please let me know if that answers your questions.